src/mc-fle2-payload-iev.c

```diff
@@ -83,6 +88,7 @@ mc_FLE2IndexedEncryptedValue_parse (mc_FLE2IndexedEncryptedValue_t *iev,
    /* Read original_bson_type. */
    CHECK_AND_RETURN (
       mc_reader_read_u8 (&reader, &iev->original_bson_type, status));
+// TODO - should we validate this, server did
```

Comment:
Is this intended to be addressed in this PR?
```suggestion
// TODO - check iev->original_bson_type is a valid BSON type
```